bolero/representation/promp_behavior.py

```diff
@@ -0,0 +1,497 @@
+        if Xdd is not None:
+            warnings.warn("Xdd is deprecated")
+        if not allow_final_velocity:
+            warnings.warn("allow_final_velocity is deprecated")
+
+        y = X.transpose(2,1,0)
+        x = np.arange(0, self.execution_time + self.dt*0.1, self.dt)
+        assert(x.shape[0] == y.shape[1])
+        sizes_ = np.array([x.shape[0]],float).repeat(y.shape[0]).flatten()
+        x_ = np.tile(x,y.shape[0]).flatten()
+        y_ = y.flatten()	
+        self.data.imitate(sizes_,x_,y_)         
+        self.y, self.yd, self.covars = self.trajectory()
+        
+
+    def trajectory(self):
+        """Generate trajectory represented by the promp in open loop.
+
+        The function can be used for debugging purposes.
+
+        Returns
+        -------
+        X : array, shape (n_steps, n_task_dims)
+            Positions
+
+        Xd : array, shape (n_steps, n_task_dims)
+            Velocities
+
+        """      
+               
+        #ret = promp.TrajectoryData(self.n_features,self.n_task_dims,True,self.overlap) #TODO make params
+        #self.data.sample_trajectory_data(ret)
+        
+        x = np.arange(0, self.execution_time + self.dt*0.1, self.dt)
+        
+        means = np.empty((self.n_task_dims*2*len(x)))
+        covars = np.empty(((self.n_task_dims*2)**2)*len(x))
+        
+        self.data.get_values(x,means,covars)
+        
+        means = means.reshape((self.n_task_dims*2,-1)).transpose(1,0)
+        covars = covars.reshape((-1,self.n_task_dims*2,self.n_task_dims*2))
+        
+        Y = means[:,::2]
+        Yd = means[:,1::2]
+        
+
+        return Y,Yd, covars
+
+    def distribution(self):
+        """Generate a trajectory's mean and covariance represented by the promp in open loop.
+
+        The function can be used for debugging purposes.
+
+        Returns
+        -------
+        means : array, shape (n_steps, n_inputs)
+            Contains positions, velocities in that order.
+            Each type is stored contiguously, i.e. for n_task_dims=2 the order
+            would be: xvxv (x: position, v: velocity).
+            
+
+        covariances : array, shape (n_steps, n_inputs,n_inputs)
+            Contains positions, velocities in that order.
+            Each type is stored contiguously, i.e. for n_task_dims=2 the order
+            would be: xvxv
+                      v
+                      x
+                      v    (x: position, v: velocity).
+        
+
+        """      
+       
+        x = np.arange(0, self.execution_time + 0.000001, self.dt)
+        
+        means = np.empty((self.n_inputs*len(x)))
+        covars = np.empty((self.n_inputs**2)*len(x))
+        self.data.get_values(x,means,covars)
+        means = means.reshape((self.n_inputs,-1)).transpose(1,0)
+        covars = covars.reshape((-1,self.n_inputs,self.n_inputs))
+               
+        
+        return means,covars
+
+    
+
+    save = save_promp_model
```

Comment:
What is this?

Reply:
that's ok: the member function save is the same as save_promp_model